### ISSUE TITLE
Feature request: "Do not ask again" checkbox when running a plugin

### ISSUE THREAD

**Issue by pgilfernandez** [synfig/synfig#3186]
**Is your feature request related to a problem? Please describe.**
While developing the "animated text" plugin, I found pretty annoying to always have to push "Proceed" button when running a plugin:
<img width="447" alt="Screenshot 2023-08-21 at 17 29 43" src="https://github.com/synfig/synfig/assets/5942369/889a4b90-0a11-4c05-b6a2-db4b33a01560">

**Describe the solution you'd like**
I would like to suggest to add a simple checkbox saying something like "do not ask again" to that popup to let users disable it.

**Describe alternatives you've considered**
none

**Additional comments**
Thanks!

**Comment by BobSynfig** [comment]
As it is written in the warning, it can't be undone and all undo history is lost...
What if you accidentally select the wrong plugin?

**Comment by pgilfernandez** [comment]
Well, if the "do not ask again" checkbox is checked that means the user knows about the warning and takes all responsibility... they couldn't blame Synfig.

BTW, it would be nice and now that I think not difficult to implement as Synfig has an "autosave" feature, to add a backup option near the autosave checkbox. With a simple field you could set backups from 0 (disabled) to any number you want. When enabled, the autosave (or the user when hits Save) creates the backup cascade with files named something like my_animation.sif1, my_animation.sif2,...

If this feature exists, even if you mark "do not ask again", accidentally select the wrong plugin and it totally break your project, you could always go back to your backup files (unless you set the backup to 0... hehehe).

**Comment by veermetri05** [comment]
I agree with @BobSynfig, it is so easy to accidentally select a wrong a plugin from the dropdown and mess your file. But whenever you run a plugin the new file isn't saved automatically. 

BTW we can press "Enter" on keyboard when we get the warning, so I don't think it impacts anybody's productivity.

If this feature is required then I recommend to put the "Do not ask again" checkbox in the preferences dialog as "Do not ask for confirmation while running plugin". So that it is hard to access and user's don't tick the "Do not ask again" checkbox accidentally, and only those aware of the risks will enable it. It increases friction, requiring the user's to locate where the option is.

**Comment by pgilfernandez** [comment]
It's not a bad idea to place it just in Preferences panel. That way probably only pro users would activate it.

BTW, if we combine this feature with the "backups" one then it would be that critical as you could always go to your backups in case of an accident.

**Comment by BobSynfig** [comment]
> That way probably only pro users would activate it.
Even pros can do mistake
But seriously, the argument of increase of productivity...
Plug-in are supposed to be rarely used, once in a while, extra functionalities that are outside of a regular workflow...
Otherwise such a functionality would be hardcoded!

**Comment by pgilfernandez** [comment]
Well, plug-ins like "text animation" intend to save you work on a daily basis, maybe new ones are coming and are used often as well... it heavily depends on the plug-in itself.

But I understand your fears, both arguments are right so, if you are still unsure go for the safest option and do not implement it till more users request it.

=)
